suspect/processing/channel_combination.py

```diff
@@ -19,3 +24,125 @@ def combine_channels(data, weights=None):
     weighted_data = weights.reshape((len(weights), 1)) * data
     combined_data = weighted_data.sum(axis=0)
     return combined_data
+    
+
+def RAMRS_SVD_weighting(data, phase_correct = True, whiten = False, W = None, noise_range = None):
+    """
+    This channel combination approach operates on raw MRS (spectral) data, 
+    prior to averaging repetitions from multiple acquisitions. It is useful in 
+    cases where there is instability in frequency and phase shifts across 
```

Comment:
This is actually the same method of calculating the channel weights, as you call the existing function. Essentially the major difference is that this function expects a 3d array instead of 2d. I think a far better solution would be to modify the existing `svd_weighting()` function so that it accepts an `axis` parameter to define which axis the channels run over, then internally it reshapes all other axes into a single line and calculates the channel weights that way. That would then be a more seamless improvement of the existing code.